Fix non-interface type ErrResourceNotFound on left

FengyunPan opened the pull request:
Related to #58145
The gophercloud.ErrResourceNotFound is not a interface, so should
use reflect to get its type then do a check.

**Release note**:
```release-note
NONE
```

FengyunPan commented:
cc @krlng

FengyunPan commented:
/assign @dims

FengyunPan commented:
/sig openstack

dims commented on pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:
```diff
@@ -1002,8 +1003,14 @@ func (lbaas *LbaasV2) ensureSecurityGroup(clusterName string, apiService *v1.Ser
 		// check whether security group does not exist
-		_, ok := err.(*gophercloud.ErrResourceNotFound)
-		if ok {
+		// The security group does not exist while 'err' is object of gophercloud.ErrResourceNotFound
```
maybe we can use a small method? since we are using the same thing 3 times.

FengyunPan replied:
Yeah, good point.

FengyunPan2 committed:
Fix non-interface type ErrResourceNotFound on left

Related to #58145
The gophercloud.ErrResourceNotFound is not a interface, so should
use reflect to get its type then do a check.

FengyunPan commented:
/test pull-kubernetes-e2e-kops-aws

FengyunPan commented:
/test pull-kubernetes-e2e-kops-aws

FengyunPan commented:
/test pull-kubernetes-e2e-kops-aws

dims commented:
/approve
/lgtm
